mne/time_frequency/psd.py

```diff
@@ -112,12 +111,208 @@ def _check_nfft(n, n_fft, n_overlap):
+
+    # Combining/reshaping to original data shape
+    psds = np.concatenate(f_psd, axis=0)
+    psds = psds.reshape(np.hstack([dshape, -1]))
+    return psds, freqs
+
+
 @verbose
+def psd_welch(inst, fmin=0, fmax=np.inf, tmin=None, tmax=None, n_fft=256,
+              n_overlap=0, picks=None, proj=False, n_jobs=1, verbose=None):
+    """Compute the PSD using Welch's method.
+
+    Parameters
+    ----------
+    inst : instance of Epochs or Raw
+        The data for PSD calculation
+    fmin : float
+        Min frequency of interest
+    fmax : float
+        Max frequency of interest
+    tmin : float | None
+        Min time of interest
+    tmax : float | None
+        Max time of interest
+    n_fft : int
+        The length of the tapers ie. the windows. The smaller
+        it is the smoother are the PSDs. The default value is 256.
+        If ``n_fft > len(inst.times)``, it will be adjusted down to
+        ``len(inst.times)``.
+    n_overlap : int
+        The number of points of overlap between blocks. Will be adjusted
+        to be <= n_fft.
+        bandwidth : float
+        The bandwidth of the multi taper windowing function in Hz.
+    picks : array-like of int | None
+        The selection of channels to include in the computation.
+        If None, take all channels.
+    proj : bool
+        Apply SSP projection vectors. If inst is ndarray this is not used.
+    n_jobs : int
+        Number of CPUs to use in the computation.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    psds : ndarray, shape ([n_epochs], n_channels, n_freqs)
+        The power spectral densities. If Raw is provided,
+        then psds will be 2-D.
+    freqs : ndarray (n_freqs)
+        The frequencies.
+    """
+    # Prep data
+    data, sfreq = _check_psd_data(inst, tmin, tmax, picks, proj)
+    psds, freqs = _psd_welch(data, sfreq, fmin=fmin, fmax=fmax, n_fft=n_fft,
+                             n_overlap=n_overlap, n_jobs=n_jobs,
+                             verbose=verbose)
+    return psds, freqs
+
+
+def _psd_multitaper(x, sfreq, fmin=0, fmax=np.inf, bandwidth=None,
+                    adaptive=False, low_bias=True, normalization='length',
+                    n_jobs=1, verbose=None):
+    """Helper function for calculating Multitaper PSD."""
+    from .multitaper import multitaper_psd
+    dshape = x.shape[:-1]
+    x = x.reshape(np.product(dshape), -1)
+
+    # Stack data so it's treated separately
+    psds, freqs = multitaper_psd(x=x, sfreq=sfreq, fmin=fmin, fmax=fmax,
+                                 bandwidth=bandwidth, adaptive=adaptive,
+                                 low_bias=low_bias,
+                                 normalization=normalization, n_jobs=n_jobs,
+                                 verbose=verbose)
+
+    # Combining/reshaping to original data shape
+    psds = psds.reshape(np.hstack([dshape, -1]))
```

Comment:
well `dshape` is going to be whatever the first N-1 dimensions are, and I need to pass a single list/tuple for the new shape, so not sure how I could do this without appending the -1. I chose hstack just because it seemed more explicit